src/scripts/main.js

```diff
@@ -1,7 +1,129 @@
 'use strict';
 
-// Uncomment the next lines to use your game instance in the browser
-// const Game = require('../modules/Game.class');
-// const game = new Game();
+function keyListener(e) {
+  if (game.getStatus() !== 'playing') {
+    return;
+  }
 
-// Write your code here
+  switch (e.key) {
+    case 'ArrowDown':
+      game.moveDown();
+      break;
+    case 'ArrowLeft':
+      game.moveLeft();
+      break;
+    case 'ArrowRight':
+      game.moveRight();
+      break;
+    case 'ArrowUp':
+      game.moveUp();
+      break;
+  }
+
+  update();
+}
+
+function update() {
+  game.getState().forEach((row, i) => {
+    row.forEach((el, j) => {
+      let cellCssClass = 'field-cell';
+
+      cellCssClass += el >= 2 ? ` field-cell--${el}` : '';
+      gameCells[i][j].innerText = el || '';
+      gameCells[i][j].className = cellCssClass;
+    });
+  });
+
+  scoreElement.innerText = game.getScore();
+
+  switch (game.getStatus().toUpperCase()) {
+    case 'WIN':
+      changeMessage('WIN');
+      break;
+    case 'LOSE':
+      changeMessage('LOSE');
+      break;
+    case 'NO_MOVES':
+      changeMessage('NO_MOVES');
+      break;
+    default:
+      hideAllMessages();
+      break;
+  }
+}
+
+function start() {
+  switch (game.getStatus()) {
+    case 'idle':
+      game.start();
+      toggleButton('restart');
+      removeStartMessage();
+      break;
+    case 'playing':
+    case 'win':
+    case 'lose':
+      game.restart();
+      toggleButton('start');
+      changeMessage('START');
+      break;
+  }
+
+  update();
+}
+
+function removeStartMessage() {
+  const startMessage = document.querySelector('.message-start');
+
+  if (startMessage) {
+    startMessage.remove();
+  }
+}
+
+function toggleButton(text) {
+  button.classList.toggle('start');
+  button.classList.toggle('restart');
+  button.innerText = text.toUpperCase();
```

Comment:
The `toggleButton` function assumes that the button will only be either 'start' or 'restart'. This could be a problem if more buttons or states are added in the future. Consider a more flexible approach for toggling button states.